src/index.js

```diff
@@ -0,0 +1,169 @@
+        taskName.nodeValue = `   ${inputField.value.trim()}`;
+        listItem.removeChild(inputField);
+        listItem.appendChild(checkbox);
+        listItem.appendChild(document.createTextNode('   '));
+        listItem.appendChild(taskName);
+        editButton.style.display = 'none';
+        moreButton.style.display = 'inline-block';
+      }
+      saveData();
+    });
+
+    // Replace the task name with the input field
+    listItem.removeChild(taskName);
+    listItem.appendChild(inputField);
+    inputField.focus();
+    saveData();
+  });
+
+  moreButton.textContent = '•••';
+  moreButton.style.float = 'right';
+  moreButton.style.cursor = 'pointer';
+  moreButton.style.backgroundColor = 'transparent';
+  moreButton.style.border = 'none';
+  moreButton.style.marginRight = '10px';
+  moreButton.addEventListener('click', () => {
+    editButton.style.display = 'inline-block';
+    moreButton.style.display = 'none';
+    saveData();
+  });
+
+  // Add the more button and edit button to the list item
+  listItem.appendChild(moreButton);
+  listItem.appendChild(editButton);
+  editButton.style.cursor = 'pointer';
+  editButton.style.display = 'none';
+
+  // Add the edit button to the list item
+  listItem.appendChild(editButton);
+
+  // Add the list item to the todo list
+  todoList.appendChild(listItem);
+
+  // Add an hr element below the new list item
+  const hr = document.createElement('hr');
+  todoList.appendChild(hr);
+
+  // Clear the input field
+  input.value = '';
+
+  // Clear the error message
+  errorMessage.textContent = '';
+  saveData();
+};
+
+const input = document.getElementById('type-todo');
+input.addEventListener('keydown', (event) => {
+  // Check if the pressed key is the "Enter" key
+  if (event.key === 'Enter') {
+    // Prevent the default behavior of the "Enter" key, which is to submit the form
+    event.preventDefault();
+
+    // Call the add task
+    addTask();
+    saveData();
+  }
+});
+
+// Add an event listener to the Add button
+const addButton = document.querySelector('.add');
+addButton.addEventListener('click', addTask);
+
+const clearCompletedTasks = () => {
+  const todoList = document.getElementById('todo-list');
+  const tasks = todoList.getElementsByTagName('li');
+
+  // Loop through all the tasks in the todo list
+  for (let i = tasks.length - 1; i >= 0; i -= 1) {
+    const task = tasks[i];
+
+    // If the task is completed (i.e., its checkbox is checked)
+    if (task.querySelector("input[type='checkbox']").checked) {
+      // Remove the task and the hr element after it
+      const hr = task.nextElementSibling;
+      todoList.removeChild(task);
+      if (hr && hr.tagName === 'HR') {
+        todoList.removeChild(hr);
+      }
+    }
+  }
+  saveData();
+};
+
+const clearButton = document.getElementById('submit');
+clearButton.addEventListener('click', clearCompletedTasks);
+
+function showTask() {
+  todoList.innerHTML = localStorage.getItem('data');
+}
+
+showTask();
```

Comment:
- [ ] Based on the project requirement you should've created a new file for this functionality 
![image](https://github.com/uno36/To-do-list/assets/62705947/da18c008-49ae-4339-a5b4-42933bed144b)